src/gui/FileBrowser.cpp

```diff
@@ -992,16 +1007,22 @@ bool FileBrowserTreeWidget::openInNewSampleTrack(FileItem* item)
 
 
 
+void FileBrowserTreeWidget::openDirectory(Directory* directory) {
+   QDesktopServices::openUrl(QUrl::fromLocalFile(directory->fullName()));
+}
+
 void FileBrowserTreeWidget::openContainingFolder(FileItem* item)
 {
-	// Delegate to QDesktopServices::openUrl with the directory of the selected file. Please note that
-	// this will only open the directory but not select the file as this is much more complicated due
-	// to different implementations that are needed for different platforms (Linux/Windows/MacOS).
-
-	// Using QDesktopServices::openUrl seems to be the most simple cross platform way which uses
-	// functionality that's already available in Qt.
 	QFileInfo fileInfo(item->fullName());
-	QDesktopServices::openUrl(QUrl::fromLocalFile(fileInfo.dir().path()));
+
+	if (FileManagerServices::canSelect())
+	{
+		FileManagerServices::select(fileInfo);
+	} else
+	{
+		QString path = QDir::toNativeSeparators(fileInfo.canonicalFilePath());
+		QDesktopServices::openUrl(QUrl::fromLocalFile(QFileInfo(path).absolutePath()));
+	}
```

Comment:
Formatting. Also, theres some unnecessary duplication. I'm not too fond of the near empty functions like `openDirectory`  that just call `QDesktopServices::openUrl`. It seems like I could call `openDirectory` in the `else` branch of this code actually. I would just remove `openDirectory` IMO and use `QDesktopServices::openUrl` directly.

```suggestion
	if (FileManagerServices::canSelect())
	{
		FileManagerServices::select(fileInfo);
	}
	else
	{
		QString path = QDir::toNativeSeparators(fileInfo.canonicalFilePath());
		QDesktopServices::openUrl(QUrl::fromLocalFile(QFileInfo(path).absolutePath()));
	}
```